src/nwb_conversion_tools/tools/hdmf.py

```diff
@@ -1,7 +1,58 @@
 from typing import Tuple
 
 import numpy as np
-from hdmf.data_utils import GenericDataChunkIterator
+from hdmf.data_utils import GenericDataChunkIterator as HDMFGenericDataChunkIterator
+
+
+class GenericDataChunkIterator(HDMFGenericDataChunkIterator):
+    def _get_default_buffer_shape(self, buffer_gb: float = 1.0) -> Tuple[int]:
+        chunk_bytes = np.prod(self.chunk_shape) * self.dtype.itemsize
+        assert buffer_gb > 0, f"buffer_gb ({buffer_gb}) must be greater than zero!"
+        assert (
+            buffer_gb >= chunk_bytes / 1e9
+        ), f"buffer_gb ({buffer_gb}) must be greater than the chunk size ({chunk_bytes / 1e9})!"
+        assert all(
+            np.array(self.chunk_shape) > 0
+        ), f"Some dimensions of chunk_shape ({self.chunk_shape}) are less than zero!"
+
+        maxshape = np.array(self.maxshape)
+
+        # Early termination condition
+        if np.prod(maxshape) * self.dtype.itemsize / 1e9 < buffer_gb:
+            return tuple(self.maxshape)
+
+        chunks_per_axis = np.ceil(maxshape / self.chunk_shape)
+        buffer_shape = np.array(self.chunk_shape)
+        buffer_bytes = chunk_bytes
+        axis_sizes_bytes = maxshape * self.dtype.itemsize
+        chunk_size_order = np.argsort(self.chunk_shape)
```

Comment:
```
        chunk_size_order = np.argsort(self.chunk_shape)
```
I would like to avoid the double indexing below and use something like:

```

index_of_smallest, index_of_second_smallest, *_ = np.argsort(x)